src/sentry/post_process_forwarder/post_process_forwarder.py

```diff
@@ -142,18 +142,48 @@ class PostProcessForwarderStrategyFactory(ProcessingStrategyFactory[KafkaPayload
     def _dispatch_function(self, message: Message[KafkaPayload]) -> None:
         raise NotImplementedError()
 
-    def __init__(self, concurrency: int):
-        self.__concurrency = concurrency
-        self.__max_pending_futures = concurrency + 1000
+    def __init__(
+        self,
+        mode: str,
+        num_processes: int,
+        input_block_size: int,
```

Comment:
all of those arguments are now optional to the strategy. i would recommend to leave them at defaults of `None`, and change the CLI (of all consumers) to accept that default

Reply:
Not sure what you meant here. The multiprocessing options [here](https://github.com/getsentry/sentry/blob/d87bc5a8bdde9feb2ec24e57838c99b9b2266f3b/src/sentry/consumers/__init__.py#L34-L54) passes the default values to the constructor. And the call made from the new unified constructor passes all the parameters to the constructor from [here](https://github.com/getsentry/sentry/blob/d87bc5a8bdde9feb2ec24e57838c99b9b2266f3b/src/sentry/consumers/__init__.py#L34-L54). Wouldn't the logic of resolving which click option to be passed already be handled by existing code? Or am I missing some context?

Reply:
I am following up on this in https://github.com/getsentry/sentry/pull/61054